classes/local/factor/object_factor_base.php

```diff
@@ -433,4 +449,33 @@ public function delete_factor_for_user($user) {
         $event = \tool_mfa\event\user_deleted_factor::user_deleted_factor_event($user, $USER, $this->name);
         $event->trigger();
     }
+
+    /**
+     * Increments the lock counter for a factor.
+     *
+     * @return void
+     */
+    public function increment_lock_counter() {
+        global $DB, $USER;
+
+        $this->lockcounter++;
+        // Update record in DB.
+        $DB->set_field('tool_mfa', 'lockcounter', $this->lockcounter, ['userid' => $USER->id, 'factor' => $this->name]);
```

Comment:
should match revoke exclusions above if we add them

Reply:
See above reply